### ISSUE TITLE
Hex edge utilities

### ISSUE THREAD

**Issue by eranimo** [flauwekeul/honeycomb#58]
Utilities for getting the edges of hexes would be appreciated. Maybe looping over them too.

**Comment by flauwekeul** [comment]
There's a function to get the [corners](https://github.com/flauwekeul/honeycomb#corners) of a hex, those are also the start points and end points of the edges. If you want the corners of all hexes in a grid, you have to translate each corner point to that of each hex. You can see this in the [PixiJS example](https://github.com/flauwekeul/honeycomb#with-pixijs).

Does this answer your question?

**Comment by flauwekeul** [comment]
Interesting. I'm working on a rewrite of Honeycomb in typescript and adding edge-related functionality sounds like a good idea. I probably won't add it in the current version though.

Thanks!

**Comment by eranimo** [comment]
Does the new v4 branch have this?

**Comment by flauwekeul** [comment]
I have an idea for an API (pseudo code):

```ts
// an edge would probably just be an interface, I don't think it needs any state
interface Edge {
  [0]: Point
}

const hexPrototype = createHexPrototype(...)
const grid = new Grid(hexPrototype, rectangle({ width: 10, height: 10 })) // creates a 10 x 10 rectangular grid

// getting a single edge requires hex coordinates and a direction:
grid.getEdge({ q: 1, r: 2 }, Compass.SE) // []
```

**Comment by eranimo** [comment]
No, as far as I can tell this is the only hex grid utils library. I think corners would be OK, as long as you can still iterate over the edges of a hex (and find the edges branching from a corner)

Associating state with an edge or a corner might be useful.

**Comment by hydralaser** [comment]
Did this ever get added? I'm looking for a way to avoid having seaborne units jump a peninsula. If I could assign hex edges as land or sea I could prevent movement over a north/south land edge shared by two coastal hexes.

**Comment by flauwekeul** [comment]
This hasn't been added. [This](https://www.redblobgames.com/grids/parts/#hexagons) may help you to get started.

**Comment by hydralaser** [comment]
Hey thanks, I hadnt seen that because I was only looking at the hexagon part of redblob's site. I'll give it a red hot crack but I've only gotten as far as I have thanks to the work you put in to honeycomb so no guarantees
